java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -525,12 +543,18 @@ public ArrowFlightSqlClientHandler build() throws SQLException {
 
         if (disableCertificateVerification) {
           clientBuilder.verifyServer(false);
+        } else {
+          if (keyStorePath != null) {
+            clientBuilder.trustedCertificates(
+                ClientAuthenticationUtils.getCertificateStream(keyStorePath, keyStorePassword));
+          } else if (useSystemTrustStore && !ClientAuthenticationUtils.isLinux()) {
```

Comment:
We should have useSystemTrustStore supercede using file-based trust stores.

Reply:
So if the client wants to use his own trustStore, he should bypass it setting useSystemTrustStore as false, right?

Reply:
Yes.

Reply:
fixed